protocols/kad/src/behaviour.rs

```diff
@@ -957,7 +957,7 @@ where
         let k = self.queries.config().replication_factor.get();
         let num_beyond_k = (usize::max(k, num_between) - k) as u32;
         let expiration = self.record_ttl.map(|ttl|
-            now + Duration::from_secs(ttl.as_secs() >> num_beyond_k)
+            now + Duration::from_secs(ttl.as_secs().checked_shr(num_beyond_k).unwrap_or(0))
```

Comment:
Does it make sense to put records that are immediately expired in the store on line 987

Reply:
As far as I can tell the record would be removed by the next `PutRecordJob::poll` invocation. Maybe @romanb has a better answer.

Never the less I would advocate for addressing this in a follow up pull-request.

Reply:
Yes, expired records are regularly purged (and a `RecordStore` may of course just drop expired records at any time anyway), but it may nevertheless be a good optimisation that may also make it a bit more difficult to somewhat maliciously or accidentally (temporarily) fill a node's store with already expired records. @mxinden if you could create an issue for it, that would be great, thanks!

Reply:
See https://github.com/libp2p/rust-libp2p/pull/1496.